wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/wdlom2wom/ValueGeneratorAndConsumer.scala

```diff
@@ -0,0 +1,22 @@
+package wdl.draft3.transforms.wdlom2wom
+
+import simulacrum.typeclass
+import scala.language.implicitConversions
+
+@typeclass
+trait ValueGenerator[A] {
+  def generatedValues(a: A): Set[String]
```

Comment:
Any chance of being stricter than `String`?   If it's really a string consider using Refined to enforce a structure.

Reply:
See below comment, it's really `generatedValueNames` for the sake of linking different elements together.